variant-remapping-automation/get_custom_assembly.py

```diff
@@ -0,0 +1,210 @@
+    It adds additional contigs provided by the required_contigs function after checking that they are not already in
+    the assembly.
+    """
+    def __init__(self, assembly_accession, assembly_fasta_path, assembly_report_path, eutils_api_key=None):
+        self.assembly_accession = assembly_accession
+        self.assembly_fasta_path = assembly_fasta_path
+        self.assembly_report_path = assembly_report_path
+        self.eutils_api_key = eutils_api_key or cfg.get('eutils_api_key')
+        self.assembly_report_headers = None
+
+    @property
+    def output_assembly_report_path(self):
+        base, ext = os.path.splitext(self.assembly_report_path)
+        return base + '_custom' + ext
+
+    @property
+    def output_assembly_fasta_path(self):
+        base, ext = os.path.splitext(self.assembly_fasta_path)
+        return base + '_custom' + ext
+
+    @property
+    def assembly_directory(self):
+        return os.path.dirname(self.assembly_fasta_path)
+
+    @cached_property
+    def required_contigs(self):
+        raise NotImplementedError
+
+    @cached_property
+    def assembly_report_rows(self):
+        """Parse the assembly report and return each row as a dict."""
+        with open(self.assembly_report_path) as open_file:
+            # Parse the assembly report file to find the header then stop
+            for line in open_file:
+                if line.lower().startswith("# sequence-name") and "sequence-role" in line.lower():
+                    self.assembly_report_headers = line.strip().split('\t')
+                    break
+            reader = DictReader(open_file, fieldnames=self.assembly_report_headers, dialect=excel_tab)
+            return [record for record in reader]
+
+    @cached_property
+    def extended_report_rows(self):
+        """Provide the list of assembly report rows extended with additional ones if there are any."""
+        if self.genbank_contig_to_add:
+            extended_report_rows = copy(self.assembly_report_rows)
+            for contig_dict in self.genbank_contig_to_add:
+                row = {
+                    "# Sequence-Name": contig_dict['genbank'],
+                    "Sequence-Role": "scaffold",
+                    "GenBank-Accn": contig_dict['genbank'],
+                    'Relationship': '<>'
+                }
+                if 'refseq' in contig_dict:
+                    row['RefSeq-Accn'] = contig_dict['refseq']
+                    row['Relationship'] = '='
+                extended_report_rows.append(row)
+        else:
+            extended_report_rows = self.assembly_report_rows
+        return extended_report_rows
+
+    @cached_property
+    def genbank_contig_to_add(self):
+        genbank_contigs = set(row['GenBank-Accn'] for row in self.assembly_report_rows)
+        return [contig_dict for contig_dict in self.required_contigs if contig_dict['genbank'] not in genbank_contigs]
+
+    @staticmethod
+    def get_written_contigs(fasta_path):
```

Comment:
I found this kind of confusing, is this a standard terminology?  If not I would prefer something more descriptive like `contig_accessions_in_fasta`.

Reply:
That the results of a bad copy/paste from [here](https://github.com/EBIvariation/eva-common-pyutils/blob/77cdd8b7dd5350de2520809de125b671347867ff/ebi_eva_common_pyutils/reference/assembly.py#L212). I've changed the name here but we might have to change it there as well.